File: 01_CF_primer/problems/02_func/func.cpp
#include <cstdio>

/* x を n 乗して返す関数 (x^n = x を n 回掛けたもの) */
double power(double x, int n) {
  double p = 1.0;
  // TODO: x を n 回掛けて x^n を計算し p に求めよ (ループを書く).
  return p;
}

int main() {
  printf("2^10 = %f\n", power(2.0, 10));
  printf("3^4  = %f\n", power(3.0, 4));
  return 0;
}

File: 01_CF_primer/problems/02_func/func.f90
module power_mod
contains
  ! x を n 乗して返す関数 (x^n = x を n 回掛けたもの)
  function power(x, n) result(p)
    real(8), intent(in) :: x
    integer, intent(in) :: n
    real(8) :: p
    integer :: i
    p = 1.0d0
    ! TODO: x を n 回掛けて x^n を計算し p に求めよ (ループを書く).
  end function power
end module power_mod

program func
  use power_mod
  print "(a,f0.6)", "2^10 = ", power(2.0d0, 10)
  print "(a,f0.6)", "3^4  = ", power(3.0d0, 4)
end program func
